service/gcs/core/gcs/gcs.go

```diff
@@ -105,6 +107,20 @@ func (e *containerCacheEntry) RemoveMappedVirtualDisk(disk prot.MappedVirtualDis
 	delete(e.MappedVirtualDisks, disk.Lun)
 	return nil
 }
+func (e *containerCacheEntry) AddMappedDirectory(dir prot.MappedDirectory) error {
+	if _, ok := e.MappedDirectories[dir.Port]; ok {
+		return errors.Errorf("a mapped directory with port %d is already attached to container %s", dir.Port, e.ID)
+	}
+	e.MappedDirectories[dir.Port] = dir
+	return nil
+}
+func (e *containerCacheEntry) RemoveMappedDirectory(dir prot.MappedDirectory) error {
```

Comment:
Not sure what we want here but typically I have always in the past had removes be `safe`. IE: Remove something that isn't there is a success because in the end its still the state you want. You could log this if you think its worth debugging but that isnt a failure to me.